var/spack/repos/builtin/packages/py-schema-salad/package.py

```diff
@@ -10,28 +10,52 @@ class PySchemaSalad(PythonPackage):
     """Schema Annotations for Linked Avro Data (SALAD)"""
 
     homepage = "https://github.com/common-workflow-language/schema_salad"
-    pypi = "schema-salad/schema-salad-8.3.20221209165047.tar.gz"
+    pypi = "schema-salad/schema_salad-8.7.20241021092521.tar.gz"
 
     license("Apache-2.0")
-
+    version(
+        "8.7.20241021092521",
+        sha256="287b27adff70e55dd715bfbea18bb1a58fd73de14b4273be4038559308089cdf",
+    )
     version(
         "8.3.20221209165047",
         sha256="d97cc9a4d7c4255eb8000bcebaa8ac0d1d31801c921fd4113ab3051c1e326c7c",
     )
 
-    depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-setuptools@45:", type="build")
+    depends_on("python@3.6:", when="@8.3.20221209165047", type=("build", "run"))
+    depends_on("python@3.9:", when="@8.7.20241021092521", type=("build", "run"))
```

Comment:
Can we use ranges on the latest active version?
```suggestion
    depends_on("python@3.6:", type=("build", "run"))
    depends_on("python@3.9:", when="@8.7.20241021092521:", type=("build", "run"))
```
where the cutoff is in https://github.com/common-workflow-language/schema_salad/commit/ddee5262f2a5100acb243493b0609d3a08c09570 (and happens to be this new version).

Same thing for other dependencies.

Rationale: this allows new versions to be added more easily and doesn't require specifying the same dependency multiple times when a new version does not have a change in dependency.